### PR TITLE
TSWEB-1080 upd Mobile page

### PR DESCRIPTION
**Pull request by Valerii-Kryvokhyzha**

### CODE REVIEW

**Review by DaniilVostrikov — APPROVED** [review]

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentActions.test.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		await button.click();
+		const newPage = await driver.DriverContext.waitForEvent('page');
+		expect(newPage.url()).toContain(url);
+		await newPage.close();
+	}
+});
+
+test('Check carousel arrows in "Mobile App Development Process" container from the "Mobile App Development" page @Regression @MobileAppDev @TSWEB-696', async () => {
+	const mobileAppDevProcessContainer = driver.getByTestId(MobileDevService.MobileAppDevelopmentProcess);
+	const numberOfClicks = 3;
```
Comment (ysukhova) [inline ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentActions.test.ts:74]: there is a single usage of numberOfClicks, so there is no real need to have it a separate variable

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts [hunk 2]
```diff
@@ -104,64 +94,114 @@ test("Check block and section titles in 'What We Do' container from the 'Mobile
...
 	await expect(allSectionTitles).toHaveText(testData);
+	await expect(technologyStackContainer.getByTestId(MainSiteButtons.RequestAQuote)).toHaveText('Request a quote');
 });
 
-test("Check section titles and images in 'Our Approach to Mobile App Development Services' container from the 'Mobile App Development Service' block @Regression @MobileDevService @TSWEB-696", async () => {
+test('Check section titles, image, and CTA button in "Industry-Specific Solution" container from the "Mobile App Development" page @Regression @MobileAppDev @TSWEB-696', async () => {
+	const industrySolutionContainer = driver.getByTestId(MobileDevService.IndustrySpecificSolution);
+	const allSectionTitles = industrySolutionContainer.getByTestId(Container.SectionTitle);
+	const testData = ['User experience', 'Moving to IaaS', 'Third-party\nservices'];
+	await expect(allSectionTitles).toHaveText(testData);
```
Comment (ysukhova) [inline ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts:110]: Please separate checks from the rest of the code with an empty line

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts [hunk 3]
```diff
@@ -104,64 +94,114 @@ test("Check block and section titles in 'What We Do' container from the 'Mobile
...
+test('Check section titles and images in "Our Approach to Mobile App Development Services" container from the "Mobile App Development" page @Regression @MobileAppDev @TSWEB-696', async () => {
 	const ourApproachContainer = driver.getByTestId(MobileDevService.OurApproachToMobileAppDevServices);
 	const allSectionTitles = ourApproachContainer.getByTestId(Container.SectionTitle);
 	const testData = ['Domain\nproficiency', 'Be a part\nof the team', 'Data-driven\ndecisions'];
 
 	await expect(allSectionTitles).toHaveText(testData);
 
-	// should be fixed and uncommented as a part of @TSWEB-897
+	const bigNumberImages = ourApproachContainer.getByTestId(MainSiteImages.BigNumber);
+	const numberOfImages = 3;
```
Comment (ysukhova) [inline ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts:140]: there is a single usage of numberOfImages, so there is no real need to have it a separate variable

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts [hunk 4]
```diff
@@ -104,64 +94,114 @@ test("Check block and section titles in 'What We Do' container from the 'Mobile
...
+
+test('Check section titles, block title, and CTA button in "The Cost of Mobile App Development" container from the "Mobile App" page @Regression @MobileAppDev @TSWEB-696', async () => {
+	const theCostOfMobAppDevContainer = driver.getByTestId(MobileDevService.TheCostOfMobileAppDevelopment);
+	const allSectionTitles = theCostOfMobAppDevContainer.getByTestId(Container.SectionTitle);
+	const testData = [
+		'Mobile app development with\npre-made design concepts',
+		'Mobile app as a part\nof a complex software solution',
+		'Full-cycle mobile app\ndevelopment',
+	];
+	await expect(allSectionTitles).toHaveText(testData);
```
Comment (ysukhova) [inline ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts:195]: Please separate checks from the rest of the code with an empty line

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts [hunk 5]
```diff
@@ -104,64 +94,114 @@ test("Check block and section titles in 'What We Do' container from the 'Mobile
...
+	const WhatIndustriesWeServeContainer = driver.getByTestId(MobileDevService.WhatIndustriesWeServe);
+	const allSectionTitles = WhatIndustriesWeServeContainer.getByTestId(Container.SectionTitle);
+	const testData = [
+		'Healthcare',
+		'Transportation\nand logistics',
+		'Energy',
+		'Leisure and\nentertainment',
+		'Agriculture',
+	];
+	await expect(allSectionTitles).toHaveText(testData);
```
Comment (ysukhova) [inline ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts:129]: Please separate checks from the rest of the code with an empty line

**Review by ysukhova — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/identifiers/MainSite/MainSiteImages.ts [hunk 6]
```diff
@@ -6,4 +6,7 @@ export default class MainSiteImages {
 	static SensorsScheme = 'Img-SensorsScheme';
 	static OurFeaturedCaseStudy = 'Img-OurFeaturedCaseStudy';
 	static SchemaCaseStudy = 'Img-SchemaCaseStudy';
+	static MobileCaseStudy = 'Img-MobileCaseStudy';
+
```
[the remark this answers is not in the document]
Comment (Valerii-Kryvokhyzha) [reply ui_tests_playwright/automation/identifiers/MainSite/MainSiteImages.ts:10]: Good point, but it's id not for specific img, but for Big numbers on site, which represented as images.
I think it little bit different from img. But if you think that need to remove this brake - i 'll fix it

image

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentActions.test.ts (hunk 1 above)
Comment (Valerii-Kryvokhyzha) [reply ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentActions.test.ts:74]: But in that way easier to understand for what this number need, without going inside a method. Should I remove this variable?

File: ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts (hunk 3 above)
Comment (Valerii-Kryvokhyzha) [reply ui_tests_playwright/automation/test/ui/MainSite/services/MobileDevService.ts/MobileAppDevelopmentContent.test.ts:140]: The same to numberOfclicks. I think in this way more transparent. If need - i'll fix

**Review by ysukhova — APPROVED** [review]

**Pull request merged**